src/arrayMethodJoin.js

```diff
@@ -4,9 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let separatorString;
+    let finalString = '';
+
+    if (separator !== ('null' || 'undefined')) {
+      separatorString = separator;
+    }
```

Comment:
```suggestion
    let finalString = '';
```
Why do you need this logic? Just use `separator` instead of `separatorString`